bitcoin/src/blockdata/block.rs

```diff
@@ -275,38 +279,48 @@ impl Block {
         merkle_tree::calculate_root(hashes).map(|h| h.into())
     }
 
-    /// base_size == size of header + size of encoded transaction count.
-    fn base_size(&self) -> Weight {
-        Weight::from_wu_usize(80 + VarInt::from(self.txdata.len()).len())
+    /// Returns the weight of the block.
+    ///
+    /// > Block weight is defined as Base size * 3 + Total size.
+    pub fn weight(&self) -> Weight {
+        // This is the exact definition of a weight unit, as defined by BIP-141 (quote above).
+        let wu = self.base_size().saturating_mul(3).saturating_add(self.total_size());
```

Comment:
I'm still not convinced that using `saturated_mul` is better than `checked_mul`.  As a downstream consumer, I'd prefer to know when the bounds are exceeded and get an error or panic instead of silently succeeding at the saturation point.  There are a number of `checked` helper methods in [weight]()https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/src/blockdata/weight.rs#L97).

Reply:
No sane use case should ever hit this limit, if a user really wants to know they can use `base_size()` and `total_size()` themselves and use the `checked` versions.

Reply:
>As a downstream consumer, I'd prefer to know when the bounds are exceeded 

I can tell you right now, without my reading or your writing the code, that the bounds are not exceeded. Would you really like to type a ton of untested panicking extra code every time you use the API, as a reminder?

Reply:
Don't have any strong opinions on usage here, but I don't like this method returning a result/option. Any other way (including how it is done) of doing this is fine is me.